src/lib/pki/PublicKeyInfrastructure.ts

```diff
@@ -0,0 +1,84 @@
+import forge from 'node-forge'
+
+function createX509Certificate() {
+  const pki = forge.pki;
+
+  // generate a keypair and create an X.509v3 certificate
+  const keys = pki.rsa.generateKeyPair(2048);
```

Comment:
Is it wise to hardcode these things? I'm not sure but I want to have a meeting to discuss PKI usage. I want to know what you're thinking regards PKI. You can writie it into the PR comments.

Reply:
not sure how far you wanted me to go, but I've added some defaults to the cert generation method.

Reply:
If you are going to to leave some defaults, can we get the defaults either at the class parameter/function parameter level, or if there's alot, then a `config` or `defaults` module file should be created to centralize it.

Reply:
These defaults have been added to the class level as static parameters.